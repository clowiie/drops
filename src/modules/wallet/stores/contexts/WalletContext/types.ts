export interface ContextType {
  walletAddress: string
}
