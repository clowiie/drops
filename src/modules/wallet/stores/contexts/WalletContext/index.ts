import { createContext } from 'react'

import { ContextType } from './types'

const WalletContext = createContext<ContextType>({
  walletAddress: '',
})

export default WalletContext
