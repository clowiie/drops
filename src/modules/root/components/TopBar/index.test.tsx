import React from 'react'

import { render } from '@testing-library/react'

import mockComponent from 'test/utils/mockComponent'
import createContextWrapper from 'test/utils/createContextWrapper'

import WalletContext from 'modules/wallet/stores/contexts/WalletContext'

import * as TopBarType from '.'

describe('TopBar', () => {
  const ConnectWalletButtonSpy = mockComponent('modules/wallet/components/ConnectWalletButton')
  const DisconnectWalletButtonSpy = mockComponent('modules/wallet/components/DisconnectWalletButton')

  const { default: TopBar } = require('.') as typeof TopBarType

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots with connected wallet', () => {
      const { container } = render(<TopBar />, {
        wrapper: createContextWrapper(WalletContext, { walletAddress: '123456' }),
      })

      expect(container.firstChild).toMatchSnapshot()
      expect(DisconnectWalletButtonSpy).toBeCalledTimes(1)
    })

    it('should match snapshots does not connect wallet', () => {
      const { container } = render(<TopBar />, {
        wrapper: createContextWrapper(WalletContext, { walletAddress: '' }),
      })

      expect(container.firstChild).toMatchSnapshot()
      expect(ConnectWalletButtonSpy).toBeCalledTimes(1)
    })
  })
})
