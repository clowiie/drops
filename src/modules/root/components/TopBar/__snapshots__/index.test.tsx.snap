// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar snapshots should match snapshots does not connect wallet 1`] = `
<div
  class="w-full bg-white shadow-sm flex justify-between items-center px-6 py-4 sm:px-12"
>
  <span
    class="text-lg font-semibold"
  >
    Drops
  </span>
  <div />
</div>
`;

exports[`TopBar snapshots should match snapshots with connected wallet 1`] = `
<div
  class="w-full bg-white shadow-sm flex justify-between items-center px-6 py-4 sm:px-12"
>
  <span
    class="text-lg font-semibold"
  >
    Drops
  </span>
  <div />
</div>
`;
